src/shared/components/DurationSelector.tsx

```diff
@@ -37,36 +48,104 @@ const DurationSelector: FunctionComponent<Props> = ({
-      testID="duration-selector"
-      button={(active, onClick) => (
-        <Dropdown.Button
-          testID="duration-selector--button"
-          active={active}
-          onClick={onClick}
-          status={getStatus(disabled)}
-        >
-          {selected.displayText}
-        </Dropdown.Button>
-      )}
-      menu={onCollapse => (
-        <Dropdown.Menu onCollapse={onCollapse} testID="duration-selector--menu">
-          {resolvedDurations.map(({duration, displayText}) => (
+    <FlexBox direction={FlexDirection.Column}>
+      <Dropdown
+        testID="duration-selector"
+        button={(active, onClick) => (
+          <Dropdown.Button
+            testID="duration-selector--button"
+            active={active}
+            onClick={onClick}
+            status={getStatus(disabled)}
+          >
+            {customDurationClicked ? 'Custom duration' : selected.displayText}
+          </Dropdown.Button>
+        )}
+        menu={onCollapse => (
+          <Dropdown.Menu
+            onCollapse={onCollapse}
+            testID="duration-selector--menu"
+            scrollToSelected={false}
+          >
             <Dropdown.Item
-              id={duration}
-              key={duration}
-              value={duration}
-              testID={`duration-selector--${duration}`}
-              selected={duration === selectedDuration}
-              onClick={onSelectDuration}
+              value={'Custom'}
+              selected={customDurationClicked}
+              onClick={() => {
+                setCustomDurationClicked(true)
+                onSelectDuration(
+                  getCustomDurationString(
+                    customDurationValue,
+                    customDurationUnit
+                  )
+                )
+              }}
             >
-              {displayText}
+              Custom duration
             </Dropdown.Item>
-          ))}
-        </Dropdown.Menu>
+            {resolvedDurations.map(({duration, displayText}) => (
+              <Dropdown.Item
+                id={duration}
+                key={duration}
+                value={duration}
+                testID={`duration-selector--${duration}`}
+                selected={
+                  duration === selectedDuration && !customDurationClicked
+                }
+                onClick={duration => {
+                  onSelectDuration(duration)
+                  setCustomDurationClicked(false)
+                }}
+              >
+                {displayText}
+              </Dropdown.Item>
+            ))}
+          </Dropdown.Menu>
+        )}
+      />
+      {customDurationClicked && (
+        <FlexBox
+          direction={FlexDirection.Row}
+          justifyContent={JustifyContent.SpaceBetween}
+          stretchToFitWidth={true}
+          style={{marginTop: '8px'}}
+        >
+          <p style={{marginLeft: '12px'}}>Enter your custom time duration:</p>
+          <TimeInput
+            style={{width: '150px'}}
```

Comment:
I'll create variables for these styles after running this enhancement by Amy! There are a couple places I hard coded the values.